### ISSUE TITLE
Random ports and gossip

### ISSUE THREAD

**Issue by avillev** [avast/gradle-docker-compose-plugin#147]
Hello,

I have a question about random ports and gossip protocols. Let's say you use two containers A and B depending on one another and you have a client connecting to A. Using some sort of internal gossip protocol, A will tell the client that he should connect to B using an ip and a port. Problem, those ip and ports are the internal ones from A's point of view and the client will never be able to connect to B.

This sadly happens quite often (kafka+zookeeper, clusters of some sort...)

A more practical example with Kafka and zookeeper:

```yaml
version: '2'
services:
  zookeeper:
    image: zookeeper:3.4.10
    ports:
      - "2181"
    environment:
      ZOO_MY_ID: 1
  kafka:
    image: wurstmeister/kafka:0.10.2.0
    depends_on:
     - zookeeper
    ports:
      - "9092"
    environment:
      KAFKA_ADVERTISED_HOST_NAME: 127.0.0.1
      KAFKA_CREATE_TOPICS: "testtopic1:1:1"
      KAFKA_ZOOKEEPER_CONNECT: zookeeper
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
```

Here the client will connect to Kafka and will never be able to get the correct external port to reach zookeeper. It also happens with the redis cluster image grokzen/redis-cluster.

Basically, do you know a way to cope that situation, configuration wise ? And if not, do you thing there is something you could do within your plugin to solve the issue ?

**Comment by augi** [comment]
Hi,

the random port is assigned to the container at the point when the container is being started so there is IMHO no way how to pass the randomly assigned port to the container e.g. via environment variable.

The right solution is IMHO that the container and the cluster must be in the same network. So you have two choices:
1. Run the client also using Docker Compose, so in your example, you would have one more service in the yaml file.
2. Define in the yaml file that these containers should use an external network. This network should be the same as the network where the client runs. For reference, see [Configure the default network](https://docs.docker.com/compose/networking/#specify-custom-networks).

I'm afraid that this plugin cannot help with this :(
